store.queryRecord broken with query-only

### Version info

```
Ember | 3.15.0
Ember Data | 3.15.1
EmberFire | 3.0.0-rc.6
Firebase | 7.8.0
```

### Steps to reproduce

```js
model() {
  this.store.queryRecord("post", { query: ref => ref.where("tags", "array-contains", "special") });
}
```

### Expected behavior

It should query for the record and resolve either a single record or not.

### Actual behavior

It throws an exception [`"Dont specify limit on queryRecord"`](https://github.com/firebase/emberfire/blob/master/addon/adapters/firestore.ts#L160)

As far as I can tell, this is due to `queryOrRef` actually being the result of calling the passed-in `query` method which results in a `Query` that has a `limit()` method on it. I'm not sure how you'd like to resolve this one but it should be easy enough.

Ironically, this is behind an `isQuery` method which only checks to see if `queryOrRef.limit` exists, then the next line throws if `queryOrRef.limit` exists.

It seems like the right thing to do here is simply ignore any limit set by the consumer. Calling `queryOrRef.limit(1)` will set it to 1 anyway, which is the proper behavior for `queryRecord`. If you really want to help developers who set it and are perplexed as to why their `limit` is being ignored when they call `queryRecord`, I recommend using Ember's [`warn`](https://api.emberjs.com/ember/3.16/functions/@ember%2Fdebug/warn) utility like so:

```js
import { warn } from '@ember/debug';
// etc...
export default class FirestoreAdapter extends DS.Adapter.extend({
  // etc...
  queryRecord<K extends keyof ModelRegistry>(store: DS.Store, type: ModelRegistry[K], options?: QueryOptions|QueryRecordOptions) {
      return rootCollection(this, type).then((ref:firestore.CollectionReference) => {
          const queryOrRef = queryRecordOptionsToQueryFn(options)(ref);
          if (isQuery(queryOrRef)) {
            warn("Don't specify limit on queryRecord", queryOrRef._query.limit);
            return queryOrRef.limit(1).get();
  // etc...
```

Actually `ember ts:precompile` nicely catch this error if you would like to build this project:

```
addon/adapters/firestore.ts(160,21): error TS2774: This condition will always return true since the function is always defined. Did you mean to call it instead?
```

So it is a real bug.

--- a/addon/adapters/firestore.ts
+++ b/addon/adapters/firestore.ts
@@ -157,7 +157,6 @@ export default class FirestoreAdapter extends DS.Adapter.extend({
         return rootCollection(this, type).then((ref:firestore.CollectionReference) => {
             const queryOrRef = queryRecordOptionsToQueryFn(options)(ref);
             if (isQuery(queryOrRef)) {
-                if (queryOrRef.limit) { throw "Dont specify limit on queryRecord" }
                 return queryOrRef.limit(1).get();
             } else {
                 (options as any).id = queryOrRef.id;

